### PR TITLE
0.2/125 harden passport auth

### PR DESCRIPTION
**Pull request by katjam**

### CODE REVIEW

File: app/Http/Controllers/API/Auth/LoginProxy.php [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+     *
+     * @param string $email
+     * @param string $password
+     */
+    public function attemptLogin($email, $password)
+    {
+        $user = User::where('email', $email)->first();
+
+        if (!is_null($user)) {
+
```
Comment (charlesstrange2) [inline app/Http/Controllers/API/Auth/LoginProxy.php]: Phpstorm thinks this bank line needs to go. Meh.

File: app/Http/Controllers/API/Auth/LoginProxy.php [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    /**
+     * Logs out the user. We revoke access token and refresh token.
+     * Also instruct the client to forget the refresh cookie.
+     */
+    public function logout()
+    {
+        $accessToken = $this->auth->user()->token();
+
+        $refreshToken = $this->db
+            ->table('oauth_refresh_tokens')
```
Comment (charlesstrange2) [inline app/Http/Controllers/API/Auth/LoginProxy.php]: refreshToken never gets used?

File: app/Http/Controllers/API/Auth/LoginProxy.php [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+    private $apiConsumer;
+    private $auth;
+    private $cookie;
+    private $db;
+    private $request;
+    private $user;
+
+    public function __construct(Application $app, User $user) {
+        $this->user = $user;
```
Comment (katjam) [inline app/Http/Controllers/API/Auth/LoginProxy.php]: Oh that was copied from source in blog. I will fix.

File: app/Http/Controllers/API/Auth/LoginProxy.php (hunk 1 above)
Comment (katjam) [inline app/Http/Controllers/API/Auth/LoginProxy.php]: Again. Copied from the blog. Will fix.

File: app/Http/Controllers/API/Auth/LoginProxy.php (hunk 2 above)
Comment (katjam) [inline app/Http/Controllers/API/Auth/LoginProxy.php]: Hmmm I will look into that.

Comment (charlesstrange2) [inline app/Http/Controllers/API/Auth/LoginProxy.php]: I dont' know if it *needs* to be used, when you declared it i kinda expected it to appear somewhere else.

**Comment by katjam** [comment]
Better?

File: app/Http/Controllers/API/Auth/LoginProxy.php (hunk 2 above)
Comment (katjam) [inline app/Http/Controllers/API/Auth/LoginProxy.php]: Copied from the blog source... I am not 100% convinced by that guy. :)

**Pull request merged**